client/my-sites/domains/domain-management/edit-contact-info-page/bulk-edit-contact-info-page.tsx

```diff
@@ -78,6 +79,28 @@ export default function BulkEditContactInfoPage( {
 		: null;
 	const firstSelectedDomain = selectedDomains?.[ 0 ];
 
+	// If all domains have the same registration agreement then we can link directly to it. Otherwise we
+	// use `''` and the form will fall back to a support page which links to all the different agreements.
+	const allDomainsShareAgreementUrl = selectedDomains?.every(
+		( domain ) =>
+			domain.domain_registration_agreement_url ===
+			firstSelectedDomain?.domain_registration_agreement_url
+	);
+	const domainRegistrationAgreementUrl =
+		( allDomainsShareAgreementUrl && firstSelectedDomain?.domain_registration_agreement_url ) || '';
+
+	const anyDomainSupportsTransferLockOptOut =
+		selectedDomains?.some( ( domain ) => {
+			if ( ! domain.transfer_lock_on_whois_update_optional ) {
+				return false;
+			}
+
+			const registrationDatePlus60Days = moment.utc( domain.registration_date ).add( 60, 'days' );
+			const isLocked = moment.utc().isSameOrBefore( registrationDatePlus60Days );
+
+			return ! isLocked;
```

Comment:
You'll see that I have expeditiously copied this logic from here:
https://github.com/Automattic/wp-calypso/blob/e9c3251162139b32841a2989949fe377c5c67be6/client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx#L150